src/Microsoft.Dnx.Tooling/Publish/PublishProject.cs

```diff
@@ -352,95 +352,94 @@ public bool PostProcess(PublishRoot root)
 
         private bool PrunePackages(PublishRoot root)
         {
-            var resolver = new DefaultPackagePathResolver(root.TargetPackagesPath);
-
-            // Special cases (for backwards compat)
-            var specialFolders = new List<string> {
-                "native",
-                "InteropAssemblies",
-                "redist",
-                "runtimes"
-            };
-
-            if (!root.NoSource)
+            if (root.MainProjectLockFile == null)
             {
-                // 'shared' folder is build time dependency, so we only copy it when deploying with source
-                specialFolders.Add("shared");
+                return true;
```

Comment:
Maybe we should remove the return value for PrunePackages. There are no cases for which we want to report a non-successful pruning that stops further publishing progress. If there are problems, such as no lock file in the main project, just don't prune anything.